operator.go

```diff
@@ -6,10 +6,21 @@ import (
 	"text/template"
 )
 
+type encoding string
```

Comment:
I don't get your idea, to be honest.

The goal I wanted to achieve by writing my own type, was to make `utf-8` a default value. That's because if we don't specify encoding for some file, like i.e.:

```
FileMetadata{
    Path: /opt/something,
    Owner: "root:root",
    Permissions: 0700,
    Content: "foobar",
}
```

I would like to render:

```
- path: /opt/something
  owner: "root:root"
  encoding: utf-8
  permissions: 0700
  content: |
    foobar
```

instead of:

```
- path: /opt/something
  owner: "root:root"
  encoding:
  permissions: 0700
  content: |
    foobar
```

which would surely result in an error. Of course I could use if statement in template to not define encoding at all if it's empty, but my solution with custom type providing default values is IMO much better.

And that's why we need String() method, because template library is using that method for types not being a `string` (even if you just define your own string type). I've tested that, it works fine and generates good cloudconfig.

The custom type you linked seems to be unrelated to this problem. Am I wrong?